src/adapt.jl

```diff
@@ -0,0 +1,9 @@
+import Adapt
+
+Adapt.adapt_structure(
+    to::Type{<:AbstractArray},
+    context::AbstractCommsContext,
+) = context(adapt(to, device(context)))
+
+Adapt.adapt_structure(::Type{<:AbstractArray}, device::AbstractDevice) =
+    CPUSingleThreaded()
```

Comment:
We should mention somewhere that we assume that we always go to `CPUSingleThreaded` when adapting to GPU.

Reply:
Yeah, I'm going to see if we can document this.

Reply:
Documenter does support documenting overloaded methods. Done.